Core/include/Acts/Propagator/Propagator.ipp

```diff
@@ -258,11 +252,6 @@ auto Acts::Propagator<S, N>::makeState(
   StateType state{eOptions, m_stepper.makeState(eOptions.stepping),
                   m_navigator.makeState(eOptions.navigation)};
 
-  static_assert(
-      detail::propagator_stepper_compatible_with<S, StateType, N>,
-      "Step method of the Stepper is not compatible with the propagator "
-      "state");
-
```

Comment:
This can't happen anymore? I believe the idea was mainly to give a better error message. I see that the outer state is now not passed into the stepper anymore, but the stepping state could still be inconsistent I guess? In any case, the `static_assert` would have to move up. I guess with the state being a simpler type in the stepper functions, the error message might be good enough on it's own though.

Reply:
maybe I was a bit too fast to remove this. the stepper is templated on the propagator so this can only be incompatible if you create a state from a different propagator with a different stepper and then pass this state to the first propagator right? is this something we have to worry about?